### ISSUE TITLE
boost::phoenix::bind issue in c++17.

### ISSUE THREAD

**Issue by myproject24** [boostorg/smart_ptr#89]
Hi All,

I am facing these below issues when I upgrade from boost_1_73 and c++14 to boost_1_77 and c++17.
What will be the problem?
Error 1:
include/boost/utility/result_of.hpp:218:8: error: 'GB* (boost::intrusive_ptr<GB>::*)() const noexcept' is not a class, struct, or union type

Error 2:
include/boost/phoenix/core/detail/function_eval.hpp:119:21: error: no type named 'type' in 'struct boost::result_of<GB* (boost::intrusive_ptr<GB>::* const(boost::intrusive_ptr<GB>&))() const noexcept>'

Thank you.

**Comment by pdimov** [comment]
I don't think this issue has anything to do with `intrusive_ptr`. It's most likely caused by `result_of` not supporting `noexcept` qualified functions.
